x-pack/plugins/endpoint/common/generate_data.ts

```diff
@@ -83,6 +83,11 @@ const OTHER_EVENT_CATEGORIES: EventInfo[] = [
 ];
 
 interface HostInfo {
+  elastic: {
```

Comment:
Outside of the scope of this PR, but wanted to mention it: 
It feels like the types/interfaces defined here and duplicated in `types.ts` should be combined so that they come from one single source.

Reply:
> Outside of the scope of this PR, but wanted to mention it:
It feels like the types/interfaces defined here and duplicated in types.ts should be combined so that they come from one single source.

Well, some of these code was changed without my notice. In my view they are two different application that knows about each other and the use cases are different. For now it is fine to leave it separate but makes for some maintenance work.

Reply:
Yeah this type can be refactored with the types in `types.ts` soon, doesn't need to be in this PR though.

Reply:
I see the challenge now @marshallmain 😄. Unrelated to this PR, I just added a new generator method to the `EndpointDocGenerator` and attempted to reference a method/type from `endpoint/public/` from this file and got an ESLint error. Some of our stuff inside of the `/public/` folder may need to first be moved to this top-level location.